app/controllers/peer_reviews_controller.rb

```diff
@@ -157,8 +157,15 @@ def assign_groups
         return
       end
     when 'unassign'
-      deleted_count, undeleted_reviews = PeerReview.unassign(selected_reviewee_group_ids,
-                                                             reviewers_to_remove_from_reviewees_map)
+      peer_reviews_filtered = PeerReview.joins(:reviewer, :reviewee)
```

Comment:
This is better, and can be improved even further. You can access the reviewer and revewee ids directly from the query by adding a `.pluck(['reviewee.id', 'reviewee.id'])`. `.pluck` will return a nested array that contains these values, which you can then iterate over directly.

This means you don't have to define the `get_reviewee_id` and `get_reviewer_id` methods (these are more inefficient because they will result in individual queries for each peer review; an instance of the "N+1 query problem").

Reply:
This [n+1 query problem](https://planetscale.com/blog/what-is-n-1-query-problem-and-how-to-solve-it) is so interesting! I wish there was another meeting to present it at 😢 

I hope CSC343 teaches similar concepts